Stairwell Integration v2.0.0 - Major Release with Breaking Changes (#42894)
demisto/content: Packs/Stairwell/ReleaseNotes/2_0_0.md

## Stairwell Integration v2.0.0

### Breaking Changes

> - **Integration ID Changed**: `Inception` → `Stairwell`

Reviewer: ```suggestion
- Updated the integration ID from **Inception** to **Stairwell**. 
```

- **Command Names Changed**: All commands now use `stairwell-*` prefix instead of `inception-*`
  - `inception-variant-discovery` → `stairwell-variant-discovery`

>   - `inception-file-enrichment` → `stairwell-file-enrichment`

Reviewer: ```suggestion
    - Changed ***inception-file-enrichment*** to ***stairwell-file-enrichment***.
```

  - `inception-metadata-get` → `stairwell-metadata-get`
  - And all other commands have been renamed with the new prefix

> - **Context Paths Changed**: `Inception.*` → `Stairwell.*`

Reviewer: ```suggestion
- Updated all context paths from *Inception* to *Stairwell*.
```

- **Display Name Changed**: "Stairwell Inception" → "Stairwell"

⚠️ **Migration Required**: Existing playbooks and scripts using the Inception integration must be updated to use the new command names and context paths.

### New Features
- **Network Intelligence**: New commands for network-based threat intelligence
- **YARA Rules**: Search and management of YARA rules

> - **Asset Management**: Commands for managing assets and asset groups

Reviewer: ```suggestion
- Added commands for managing assets and asset groups.
```

- **Intake System**: File upload and processing with detonation support
  - Upload files from War Room, URLs, or file paths
  - Automatic file analysis and processing

>   - Detonation plan support

Reviewer: ```suggestion
  - Added support for detonation plans.
```

  - Origin tracking (web referrers, zones)
- **Object Operations**: Sightings tracking, detonation triggers, opinions management

>   - `stairwell-object-sightings` - List sightings for an object

Reviewer: ```suggestion
    - Added the ***stairwell-object-sightings*** command that lists sightings for an object.
```

  - `stairwell-object-detonation-trigger` - Trigger detonation

>   - `stairwell-object-detonation-get` - Get detonation results

Reviewer: ```suggestion
    - Added the ***stairwell-object-detonation-get*** command that gets detonation results.
```

  - `stairwell-object-opinion-list` - List opinions for an object

> - **Enhanced Metadata**: Comprehensive metadata extraction and analysis

Reviewer: ```suggestion
- Added support for comprehensive metadata extraction and analysis.
```

### Improvements

> - **Enhanced Error Handling**: Significantly improved error messages across all commands

Reviewer: - Improved the error messages across all commands.

  - Added timeout exception handling with clear, actionable messages

>   - Protected type conversion operations to prevent crashes from invalid input

Reviewer: ```suggestion
  - Improved type conversion operations to prevent crashes from invalid inputs.
```

  - Added missing exception handling to object-related commands (sightings, detonation, opinions)
  - Improved error messages with context and troubleshooting guidance
- **Better HTTP Error Reporting**: Standardized HTTP status code error messages
  - 401: Clear message about API key authentication
  - 404: Specific message about missing resources
  - 429: Rate limit guidance

>   - 500-504: Server/gateway error guidance

Reviewer: ```suggestion
```

- **Improved Debugging**: Enhanced logging for file cleanup operations
- **Input Validation**: Better validation for numeric parameters (pageSize, originZoneId)

### Bug Fixes
- Fixed potential crashes from timeout exceptions in file upload and download operations

> - Fixed potential crashes from invalid page_size parameters in YARA and Asset commands

Reviewer: ```suggestion
- Fixed an issue where invalid page_size parameters caused potential crashes in the YARA and Asset commands.
```
